genesyscloud/telephony_providers_edges_phone/genesyscloud_telephony_providers_edges_phone_proxy.go

```diff
@@ -0,0 +1,270 @@
+}
+
+// unassignUserFromStation unassigns a user from the station
+func (p *phoneProxy) unassignUserFromStation(ctx context.Context, stationId string) (*platformclientv2.APIResponse, error) {
+	return p.unassignUserFromStationAttr(ctx, p, stationId)
+}
+
+// assignUserToStation assigns a user to the station
+func (p *phoneProxy) assignUserToStation(ctx context.Context, userId string, stationId string) (*platformclientv2.APIResponse, error) {
+	return p.assignUserToStationAttr(ctx, p, userId, stationId)
+}
+
+// getAllPhonesFn is an implementation function for retrieving all Genesys Cloud Phones
+func getAllPhonesFn(ctx context.Context, p *phoneProxy) (*[]platformclientv2.Phone, error) {
+	var allPhones []platformclientv2.Phone
+
+	phones, _, _ := p.edgesApi.GetTelephonyProvidersEdgesPhones(1, 100, "", "", "", "", "", "", "", "", "", "", "", "", "", nil, nil)
+	for _, phone := range *phones.Entities {
+		if phone.State != nil && *phone.State != "deleted" {
+			allPhones = append(allPhones, phone)
+		}
+	}
+
+	for pageNum := 2; pageNum <= *phones.PageCount; pageNum++ {
+		const pageSize = 100
+		const sortBy = "id"
+		phones, _, err := p.edgesApi.GetTelephonyProvidersEdgesPhones(pageNum, pageSize, sortBy, "", "", "", "", "", "", "", "", "", "", "", "", nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if phones.Entities == nil || len(*phones.Entities) == 0 {
+			break
+		}
+
+		for _, phone := range *phones.Entities {
+			if phone.State != nil && *phone.State != "deleted" {
+				allPhones = append(allPhones, phone)
+			}
+		}
+	}
+
+	return &allPhones, nil
+}
+
+// createPhoneFn is an implementation function for creating a Genesys Cloud Phone
+func createPhoneFn(ctx context.Context, p *phoneProxy, phoneConfig *platformclientv2.Phone) (*platformclientv2.Phone, *platformclientv2.APIResponse, error) {
+	phone, resp, err := p.edgesApi.PostTelephonyProvidersEdgesPhones(*phoneConfig)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return phone, resp, nil
+}
+
+// getPhoneByIdFn is an implementation function for retrieving a Genesys Cloud Phone by id
+func getPhoneByIdFn(ctx context.Context, p *phoneProxy, phoneId string) (*platformclientv2.Phone, *platformclientv2.APIResponse, error) {
+	phone, resp, err := p.edgesApi.GetTelephonyProvidersEdgesPhone(phoneId)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return phone, resp, nil
+}
+
+// getPhoneByNameFn is an implementation function for retrieving a Genesys Cloud Phone by name
+func getPhoneByNameFn(ctx context.Context, p *phoneProxy, phoneName string) (phone *platformclientv2.Phone, retryable bool, err error) {
+	const pageNum = 1
+	const pageSize = 100
+	phones, _, err := p.edgesApi.GetTelephonyProvidersEdgesPhones(pageNum, pageSize, "", "", "", "", "", "", "", "", "", "", phoneName, "", "", nil, nil)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if phones.Entities == nil || len(*phones.Entities) == 0 {
+		return nil, true, err
+	}
+
+	return &(*phones.Entities)[0], false, nil
```

Comment:
See above comment about looping through the entities returned and doing an exact match on name.

Reply:
fixed